ParProcCo/msmapper_aggregator.py

```diff
@@ -75,63 +75,72 @@ def _get_nxdata(self):
             self.data_shape = signal_shape
-            self.axes_spacing = []
-            for axis in axes:
-                spacing = round((axis[1] - axis[0]), 6)
-                self.axes_spacing.append(spacing)
+            self.axes_spacing = [round((axis[1] - axis[0]), 6) for axis in axes]
 
             if self.renormalisation:
-                aux_signal = nxdata[self.aux_signal_name]
-                assert(len(aux_signal.shape) == self.data_dimensions)
-                assert(aux_signal.shape == self.data_shape)
+                weights = nxdata["weight"]
+                assert len(weights.shape) == self.data_dimensions
+                assert weights.shape == self.data_shape
 
     def _get_default_nxgroup(self, f: Union[h5py.File, h5py.Group], class_name: str) -> str:
         if "default" in f.attrs:
             group_name = f.attrs["default"]
+            assert f[group_name].attrs["NX_class"] == class_name, f"{group_name} should be {class_name}"
             return group_name
 
-        f_key_list = list(f.keys())
-        for key in f_key_list:
-            key_attributes = list(f[key].attrs)
-            if ("NX_class" in key_attributes) and (f[key].attrs["NX_class"] == class_name):
-                group_name = key
-                return group_name
+        for key in f.keys():
+            try:
+                if f[key].attrs.get("NX_class", '') == class_name:
+                    return key
+            except KeyError:
+                # TODO: example data is missing linked group
+                pass
         raise ValueError
 
     def _get_default_signals_and_axes(self, nxdata: h5py.Dataset) -> None:
+        self.renormalisation = False
+        self.accumulate_aux_signals = False
+
         if "auxiliary_signals" in nxdata.attrs:
-            aux_signal_names = [name.decode() for name in nxdata.attrs["auxiliary_signals"]]
-            assert(len(aux_signal_names) == 1)
-            self.aux_signal_name = aux_signal_names[0]
-            self.renormalisation = True
+            self.aux_signal_names = [name.decode() for name in nxdata.attrs["auxiliary_signals"]]
+            # TODO: add code to accumulate aux_signals
+            self.accumulate_aux_signals = False if self.aux_signal_names == ["weight"] else True
+            if "weight" in self.aux_signal_names:
+                self.renormalisation = True
         else:
-            self.aux_signal_name = None
+            self.aux_signal_names = None
+
+        if not self.renormalisation:
             self.renormalisation = False
 
         if "signal" in nxdata.attrs:
             self.signal_name = nxdata.attrs["signal"].decode()
-            if "axes" in nxdata.attrs:
-                self.axes_names = [name.decode() for name in nxdata.attrs["axes"]]
-                # TODO: add logic to create axes if not in file (from integer sequence starting at 0)
-                return
+        elif "data" in nxdata.attrs:
```

Comment:
No, see if there's a dataset called "data" in this `nxdata` group. If so then use as signal name